### DISCUSSION TITLE
Trouble with Emotion not working in SPA mode with Remix

### DISCUSSION THREAD

**Discussion by apple-yagi** [remix-run/remix#9257] [Q&A]
Hey everyone,

I've been trying to get Emotion working with Remix in SPA mode, but I've hit a roadblock. It seems like Emotion styles aren't being applied properly. I've searched around for solutions but haven't found anything conclusive.

Has anyone else encountered this issue? And if so, were you able to find a workaround or fix for it? I'd appreciate any insights or tips on how to make Emotion cooperate smoothly with Remix in SPA mode.

Thanks in advance for your help!

**Comment by vladinator1000** [comment]
I'm also interested in this. It's such a pain in the butt currently.
The only useful resource I've seen is this https://github.com/remix-run/examples/tree/main/emotion, but it comes with drawbacks
- Doesn't work with streaming
- Doesn't work with [Layout export](https://remix.run/docs/en/main/file-conventions/root#layout-export)
- It's using an old version of remix
- Wastefully renders the whole tree twice

Do you have a reproduction project @apple-yagi maybe we could collaborate on this

**Comment by suhaotian** [comment]
Try NextUI, emotion not good and have performance issue(check the HTML output).

**Reply by vladinator1000** [reply]
I think I'm just gonna go with the built-in CSS modules, the other frameworks aren't worth the hassle.
